src/recycle/shared_pool.hpp

```diff
@@ -6,6 +6,7 @@
 #pragma once
 
 #include <algorithm>
+#include <atomic>
```

Comment:
Let us not use atomics here, but rather protect the count by the lock (injected with the locking policy).